packages/dashboard/src/components/robots/robot-page.tsx

```diff
@@ -14,13 +14,37 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+// custom hook for auto refresh
+function useAutoRefresh(callback: () => Promise<any>, delay: number, autoRefresh: boolean) {
+  const savedCallback = React.useRef<() => Promise<any>>();
+
+  React.useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  React.useEffect(() => {
+    let intervalId: NodeJS.Timeout;
+    const ticker = () => {
+      savedCallback.current && savedCallback.current();
+    };
+    if (autoRefresh) {
+      intervalId = setInterval(ticker, delay);
+      return () => clearInterval(intervalId);
+    } else {
+      return;
+    }
+  }, [delay, autoRefresh]);
+}
+
```

Comment:
Is this functionally same as `setInterval` inside a `React.useEffect`?

---

`savedCallback` is not in the dependencies of the effect so you would get bugs if you change the callback. The use of `savedCallback` is also questionable, you could `useCallback` instead.